src/transports/ssh.c

```diff
@@ -893,9 +913,24 @@ int git_transport_ssh_with_paths(git_transport **out, git_remote *owner, void *p
 #endif
 }
 
+void git_transport_ssh_global_shutdown(void);
+
 int git_transport_ssh_global_init(void)
 {
 #ifdef GIT_SSH
+#ifdef GIT_WIN32
+	const char *(*libssh2_version)(int) = NULL;
+#else
+	libssh2_handle = dlopen("libssh2.dylib", RTLD_LAZY);
+	const char *(*libssh2_version)(int) = dlsym(libssh2_handle, "libssh2_version");
+#endif
+
+	// Check that libssh2's version is sufficient for us
+	if (!libssh2_version || !libssh2_version(LIBSSH2_VERSION_NUM)) {
+		return 0;
+	}
```

Comment:
Delaying the error message until first use seems reasonable. In that case I'd suggest stashing away an appropriate error message about what exactly failed during init. So the user can get something more useful than just ` "Cannot create SSH transport. SSH support unavailable"`. I'm guessing it will then be either something like "Cannot create SSH transport. SSH support unavailable. Failed loading libssh2.dylib" or "Cannot create SSH transport. SSH support unavailable.  libssh2.dylib is version XYZ which is not compatible with version PQR".

I'd also pull the checks out into their own function, so that we can do something nicer if libssh2 is statically linked.
(I assume that is planned to remain an option?). At the moment it looks like if its statically linked, it will still try to `dlopen` then `.dylib` and then report errors about not having ssh.